### PR TITLE
Add classify support for http-proxy

### PR DESCRIPTION
**Pull request by yupbank**
Right now, the default tf-model exported example is using classify api, this pr add classify support.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/341)

### CODE REVIEW

File: components/k8s-model-server/http-proxy/server.py [hunk 1]
```diff
@@ -129,7 +188,9 @@ def get_application(**settings):
     return tornado.web.Application(
             [
             (r"/model/(.*):predict", PredictHandler),
+            (r"/model/(.*):classify", ClassifyHandler),
```
Comment (jlewi) [inline components/k8s-model-server/http-proxy/server.py]: Can you add docs explaining the difference between predict and classify?

Comment (yupbank) [reply components/k8s-model-server/http-proxy/server.py]: so with classify method, the default model we have `gs://kubeflow-models/inception`  would finally work with http-proxy.

File: components/k8s-model-server/http-proxy/server.py [hunk 2]
```diff
@@ -86,7 +122,12 @@ def decode_b64_if_needed(value):
     else:
         return value
 
+
 class PredictHandler(tornado.web.RequestHandler):
+    """
+    Predict Hanlder proxy predict method, the input of tf savedModel is expected to be a `Map<strinbg, tf.Tensor>` protobuf
```
Comment (jlewi) [inline components/k8s-model-server/http-proxy/server.py]: What does protobuf mean in this context? i.e. does it mean the json representation of the protobuf?

File: components/k8s-model-server/http-proxy/server.py (hunk 1 above)
Comment (jlewi) [reply components/k8s-model-server/http-proxy/server.py]: So the difference is ClassifyHandler expects as input a json dictionary with fields model_spec and input where as PredictHandler expects a map of names to tensor?

Can you update README.md to explain this?

**Comment by yupbank** [comment]
> So the difference is ClassifyHandler expects as input a json dictionary with fields model_spec and input where as PredictHandler expects a map of names to tensor?

The different is actually implementation details, from the API we expose to user, it's the same...

Just that, when the user is exporting their model they have options to export as classify or predict.

https://github.com/tensorflow/serving/issues/794 
i have filed an issue upstream 

> Can you update README.md to explain this?

I will add one!

File: components/k8s-model-server/http-proxy/server.py (hunk 2 above)
Comment (yupbank) [reply components/k8s-model-server/http-proxy/server.py]: the protobuf is referring to backend.. from the user perspective, the input schema is the same

**Comment by jlewi** [comment]
/lgtm

More documentation would be useful but I don't think we need to block this PR for that.

@lluunn Could you please take a quick look since you are the prediction expert?

**Pull request merged**